pkg/mesh-networking/apply/configtarget/config_target_validator.go

```
@@ -187,6 +189,56 @@ func validateOneVirtualMeshPerMesh(virtualMeshes []*v1.VirtualMesh) {
 			}
 			invalidVirtualMeshes.Insert(vMesh)
 		}
+
+		if len(vMesh.Spec.GetFederation().GetIngressGatewayServiceSelectors()) > 0 {
+			// Validate Ingress gateway service selectors
+			matchedDestinations := 0
+			for _, ingressGatewayServiceSelector := range vMesh.Spec.GetFederation().GetIngressGatewayServiceSelectors() {
+				gatewayTlsPortName := "tls"
+				if ingressGatewayServiceSelector.GetGatewayTlsPortName() != "" {
+					gatewayTlsPortName = ingressGatewayServiceSelector.GetGatewayTlsPortName()
+				}
+				for _, destination := range destinations.List() {
+					if selectorutils.SelectorMatchesDestination(ingressGatewayServiceSelector.GetDestinationSelectors(), destination) {
```

We also need to validate that Destinations selected by direct reference exist

---

That should be checked in the `SelectorMatchesDestination` function

---

to clarify, we should report an error if any direct references don't exist, [we can reuse this function](https://github.com/solo-io/gloo-mesh/blob/8d4fbb67b918dbee810d449b3ee797ca40393d8e/pkg/mesh-networking/apply/configtarget/config_target_validator.go#L102)